ibm/resource_ibm_enterprise_account.go

```diff
@@ -0,0 +1,312 @@
+/**
+ * (C) Copyright IBM Corp. 2021.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ibm
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"log"
+	"time"
+
+	"github.com/IBM/platform-services-go-sdk/enterprisemanagementv1"
+)
+
+func resourceIbmEnterpriseAccount() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceIbmEnterpriseAccountCreate,
+		ReadContext:   resourceIbmEnterpriseAccountRead,
+		UpdateContext: resourceIbmEnterpriseAccountUpdate,
+		DeleteContext: resourceIbmEnterpriseAccountDelete,
+		Importer:      &schema.ResourceImporter{},
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Update: schema.DefaultTimeout(20 * time.Minute),
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+		Schema: map[string]*schema.Schema{
+			"parent": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The CRN of the parent under which the account will be created. The parent can be an existing account group or the enterprise itself.",
+			},
+			"name": &schema.Schema{
+				Type:             schema.TypeString,
+				Optional:         true,
+				Description:      "The name of the account. This field must have 3 - 60 characters.",
+				DiffSuppressFunc: applyOnce,
```

Comment:
you added diffsupress and update function for name?

Reply:
Yes, in case of any account which is created under enterprise, only updating the parent is allowed.
Reference: https://cloud.ibm.com/apidocs/enterprise-apis/enterprise#update-account